### ISSUE TITLE
Exploration action vs execution action in MAPPO.py

### ISSUE THREAD

**Issue by Lplenka** [Derekabc/MARL_CAVs#12]
Hello @Derekabc 

Thanks for creating this repository. I have read two of your papers: "Deep Multi-agent Reinforcement Learning for Highway On-Ramp Merging in Mixed Traffic" [[arxiv](https://arxiv.org/abs/2105.05701)] and "Multi-agent Reinforcement Learning for Cooperative Lane Changing of Connected and Autonomous Vehicles in Mixed Traffic" [[arxiv](https://arxiv.org/abs/2111.06318)].

Honestly, both the papers are inspiring.

I am working on a project which uses [highway-env](https://github.com/eleurent/highway-env) environment's highway environment. I was going through your [MAPPO.py](https://github.com/Derekabc/MARL_CAVs/blob/main/MARL/MAPPO.py) code, and I am not able to understand why two different functions [exploration_action()](https://github.com/Derekabc/MARL_CAVs/blob/main/MARL/MAPPO.py#L216), and  [action()](https://github.com/Derekabc/MARL_CAVs/blob/main/MARL/MAPPO.py#L224) have same code.

As far I understood you are using one during the training and the other during evaluation. However, based on your comments above those two functions, [exploration_action()](https://github.com/Derekabc/MARL_CAVs/blob/main/MARL/MAPPO.py#L216) is supposed to be used during exploration and [action()](https://github.com/Derekabc/MARL_CAVs/blob/main/MARL/MAPPO.py#L224) is supposed to be used during exploitation phase of RL agent. But I am not sure why both the functions are exactly same.

Please help me understand your idea on this.

**Comment by DongChen06** [comment]
Hi @Lplenka , thanks for your interest. You are right, in the `action()` function, we should use the argmax to get the most likely actions to avoid having randomness, like  `actions = np.argmax(softmax_actions, axis=1)`.  Thanks for pointing it out.

**Comment by Lplenka** [comment]
@Derekabc, that makes sense, I was thinking the same. The surprising part is the model still works decently, I wonder how 🤔. Regardless, I will fix it and see how the performance improves.

**Comment by DongChen06** [comment]
@Lplenka thanks, looking forward to your progress! Also, if you are interested in our MA2C code, please send me an email.

**Comment by Lplenka** [comment]
Thanks, exploring the MA2C code would be insightful. I will connect with you by mail.
